Formula/gfan.rb

```diff
@@ -55,31 +49,284 @@ def install
+ 	r++;
+       }
+   assert(0);
++  exit(1);
+ }
+ 
+ static int sIndex(int i, int j, int N, bool withMasses)
+diff --git a/src/packedmonomial.cpp b/src/packedmonomial.cpp
+index 65a9242..37404e6 100644
+--- a/src/packedmonomial.cpp
++++ b/src/packedmonomial.cpp
+@@ -39,6 +39,7 @@ vector<MonomialType> minimized(vector<MonomialType> const &generators)
+ 
+ 	  *g=ret;
+ */
++	exit(1);
+ }
+ 
+ 
+diff --git a/src/parser.cpp b/src/parser.cpp
+index 8c09b1a..ed807f8 100644
+--- a/src/parser.cpp
++++ b/src/parser.cpp
+@@ -638,6 +638,7 @@ Field CharacterBasedParser::parseField()
+ 
+   parserError("field",c);
+   assert(0);
++  exit(1);
+ }
+ 
+ 
+diff --git a/src/polyhedralfan.cpp b/src/polyhedralfan.cpp
+index e62cae4..90eae40 100644
+--- a/src/polyhedralfan.cpp
++++ b/src/polyhedralfan.cpp
+@@ -1672,6 +1672,7 @@ PolyhedralCone PolyhedralFan::coneContaining(IntegerVector const &v)const
+     if(i->contains(v))return i->faceContaining(v);
+   debug<<"Vector "<<v<<" not contained in support of fan\n";
+   assert(0);
++  exit(1);
+ }
+ 
+ 
+diff --git a/src/polymakefile.cpp b/src/polymakefile.cpp
+index a7ea692..55d3d39 100644
+--- a/src/polymakefile.cpp
++++ b/src/polymakefile.cpp
+@@ -329,6 +329,7 @@ void PolymakeFile::writeMatrixProperty(const char *p, const IntegerMatrix &m, bo
+ IntegerMatrix PolymakeFile::readArrayArrayIntProperty(const char *p, int width)
+ {
+   assert(0);//Not implemented yet.
++  exit(1);
+ }
+ 
+ 
+@@ -371,7 +372,7 @@ static list<int> readIntList(istream &s)
+ {
+   list<int> ret;
+   int c=s.peek();
+-  while((c>='0') && (c<='9')|| (c==' '))
++  while(((c>='0') && (c<='9'))|| (c==' '))
+     {
+       //      fprintf(Stderr,"?\n");
+       int r;
+diff --git a/src/polynomialgcd.cpp b/src/polynomialgcd.cpp
+index 82aea06..14cb528 100644
+--- a/src/polynomialgcd.cpp
++++ b/src/polynomialgcd.cpp
+@@ -618,7 +618,7 @@ Polynomial NonMonomialPolynomialGCDForZ(PolynomialSet p)
+ if(1)	{
+ 		static int i;
+ 		i++;
+-		if((i==1000))
++		if(i==1000)
+ 		{
+ //			debug<<simplifyPolysViaHomogeneitySpace(p);
+ //			debug<<"NonMon on:"<<p.getRing()<<p<<"\n";
+diff --git a/src/timer.cpp b/src/timer.cpp
+index 7421c41..b5d8741 100644
+--- a/src/timer.cpp
++++ b/src/timer.cpp
+@@ -1,10 +1,10 @@
+ #include "timer.h"
+-#include "log.h"
+-
+ #include "iostream"
+ #include <time.h>
+ #include <assert.h>
+ 
++#include "log.h" // include last, because it defines a macro "log2" also defined as a function in math.h
++
+ using namespace std;
+ 
+ Timer* Timer::timers;
+diff --git a/src/vektor.cpp b/src/vektor.cpp
+index d17579a..63e39f0 100644
+--- a/src/vektor.cpp
++++ b/src/vektor.cpp
+@@ -96,7 +96,7 @@ int gcdGFAN(int r, int s)
+ int gcdOfVector(IntegerVector const &v)
+ {
+   int ret=0;
+-  for(int i=0;i<v.size();i++)if(ret=v[i])break;
++  for(int i=0;i<v.size();i++)if((ret=v[i]))break;
+   if(ret<0)ret=-ret;
+   assert(ret!=0);
+   for(int i=0;i<v.size();i++)ret=gcdGFAN(ret,v[i]);
```

Comment:
This patch contains the contents of `M2/libraries/gfan/patch-0.6.2`, right?  Do we know why these changes are necessary?  I've only needed the `<cstdint>` patch for the Debian gfan package to build.

Reply:
No, a few things have changed.

Reply:
You're right that the patches seem excessive. I'll look into whether the `exit(1)`s were necessary to fix something in the package.

Reply:
Yeah, basically Dan wanted error signals it seems:
https://github.com/Macaulay2/M2/commit/f088148555806540924ec167f13ea487615c83ee#diff-bd52196e0826f3b0c95eb09f00bc87f1bfedb2c4752f017a631ec6517d28477e

Reply:
Those `exit`'s aren't in the Debian/Ubuntu, Fedora, or Gentoo gfan packages.

Reply:
Probably the package should be updated to reflect this ...

Reply:
I don't understand the point of the `exit`'s.  They all appear in something like this:

```c
assert(0);
exit(1);
```

But `assert(0)` will always abort the program, so we never even reach the call to `exit`.

Reply:
Yeah, I think once the int128 issue is resolved I'll just remove the exits.